src/components/layout/CyVerseAppBar.js

```diff
@@ -242,10 +244,42 @@ function CyverseAppBar(props) {
         }
     }, [clientConfig, setConfig]);
 
+    useEffect(() => {
+        if (bootstrapError) {
+            const errorString = JSON.stringify(bootstrapError);
+            setBootstrapError(null);
+            router.push("/error?errorInfo=" + errorString);
+        }
+    }, [bootstrapError, router]);
+
     useEffect(() => {
         setAppBarRef(ref);
     }, [ref, setAppBarRef]);
 
+    useEffect(() => {
+        if (userProfile) {
+            const ip = userProfile.attributes.ip;
+            setBootstrapQueryKey([BOOTSTRAP_KEY, { ip }]);
+        }
+    }, [userProfile]);
+
+    useQuery({
+        queryKey: bootstrapQueryKey,
+        queryFn: bootstrap,
+        config: {
+            staleTime: Infinity,
+            cacheTime: Infinity,
+            retry: 1,
+            //copied from react-query doc. Add exponential delay for retry.
+            retryDelay: (attempt) =>
+                Math.min(attempt > 1 ? 2 ** attempt * 1000 : 1000, 30 * 1000),
+            onError: (e) => {
+                console.log("error on bootstrap=>" + e);
+                setBootstrapError(e);
+            },
+        },
+    });
```

Comment:
`prefetchQuery` was not invoking `onError` for some reason so I switched it back to `useQuery`. But I didn't notice any operational or performance differences.